lib/Runtime/Debug/TTEvents.cpp

```diff
@@ -444,18 +444,23 @@ namespace TTD
             return callEvt->LastNestedEventTime;
         }
 
-        void ExternalCallEventLogEntry_ProcessArgs(EventLogEntry* evt, int32 rootDepth, Js::JavascriptFunction* function, uint32 argc, Js::Var* argv, bool checkExceptions, UnlinkableSlabAllocator& alloc)
+        void ExternalCallEventLogEntry_ProcessArgs(EventLogEntry* evt, int32 rootDepth, Js::JavascriptFunction* function, const Js::Arguments& args, bool checkExceptions, UnlinkableSlabAllocator& alloc)
         {
             ExternalCallEventLogEntry* callEvt = GetInlineEventDataAs<ExternalCallEventLogEntry, EventKind::ExternalCallTag>(evt);
 
             callEvt->RootNestingDepth = rootDepth;
-            callEvt->ArgCount = argc + 1;
+            callEvt->ArgCount = args.Info.Count + 1 + (args.HasNewTarget() ? 1 : 0);
 
             static_assert(sizeof(TTDVar) == sizeof(Js::Var), "These need to be the same size (and have same bit layout) for this to work!");
 
             callEvt->ArgArray = alloc.SlabAllocateArray<TTDVar>(callEvt->ArgCount);
             callEvt->ArgArray[0] = static_cast<TTDVar>(function);
-            js_memcpy_s(callEvt->ArgArray + 1, (callEvt->ArgCount - 1) * sizeof(TTDVar), argv, argc * sizeof(Js::Var));
+            js_memcpy_s(callEvt->ArgArray + 1, args.Info.Count * sizeof(TTDVar), args.Values, args.Info.Count * sizeof(Js::Var));
+
+            if (args.HasNewTarget())
+            {
+                callEvt->ArgArray[args.Info.Count + 1] = static_cast<TTDVar>(args.GetNewTarget());
```

Comment:
Would it make sense to allocate one more pointer in `callEvt` to handle the `new.target` (`callEvt->newTargetPointer`?) instead of tacking it on at the end? This is our data structure so we don't really need to sneakily add an extra arg here and it might be confusing why the argument count doesn't match.

Reply:
@mrkmarron can you have a look at the change I made for this?